pandas/io/parsers/base_parser.py

```diff
@@ -1116,14 +1119,19 @@ def converter(*date_cols, col: Hashable):
                 date_format.get(col) if isinstance(date_format, dict) else date_format
             )
 
-            return tools.to_datetime(
+            result = tools.to_datetime(
                 ensure_object(strs),
                 format=date_fmt,
                 utc=False,
                 dayfirst=dayfirst,
                 errors="ignore",
                 cache=cache_dates,
-            )._values
+            )
+            if isinstance(result, DatetimeIndex):
+                arr = result.to_numpy()
```

Comment:
do we know it is timezone-naive here?

Reply:
no tz-aware.

I guess you are referring to ``result._values._ndarray``? Tried this first, but breaks tests

Reply:
its its tzaware then to_numpy() should convert to object, which i havent checked but assume we dont want here

Reply:
Yeah it does, but I am not too concerned by this since this is the same behavior as before. This keeps performance at least stable.

Reply:
I have to clarify my initial response: we can either be tz aware or naive, depends on the input

Reply:
@jbrockmendel ok with merging? We should get this into 2.0

Reply:
OK with me

Reply:
thx